pkg/collector/corechecks/ebpf/probe/ebpfcheck/probe.go

```diff
@@ -523,3 +525,80 @@ func ringBufferMemoryUsage(mapStats *model.EBPFMapStats, info *ebpf.MapInfo, k *
 	mapStats.RSS = mapStats.MaxSize
 	return nil
 }
+
+func hashMapNumberOfEntries(mp *ebpf.Map) int64 {
+	if isPerCPU(mp.Type()) {
+		return -1
+	}
+
+	numElements := 0
+	if ddebpf.BatchAPISupported() && mp.Type() != ebpf.HashOfMaps { // HashOfMaps doesn't work with batch API
+		// Here we duplicate a bit the code from cilium/ebpf to use the batch API
```

Comment:
Can you extract this block into a separate function? It would make the code a bit easier to read

Reply:
You mean the condition? The contents of the `if` are in a different function (`hashMapNumberOfEntriesWithBatch`)

Reply:
Yeah, Github was showing me outdated code. It is fine.